addons/decimal_precision/models/decimal_precision.py

```diff
@@ -13,6 +13,25 @@ class DecimalPrecision(models.Model):
         ('name_uniq', 'unique (name)', """Only one value can be defined for each given usage!"""),
     ]
 
+    @api.onchange('digits')
+    def _onchange_digits(self):
+        new_rounding = 1.0 / 10.0**self.digits
+        dangerous_uom = self.env['product.uom'].search([('rounding', '<', new_rounding)])
```

Comment:
This could miss the archived UoM

```suggestion
        dangerous_uom = self.env['product.uom'].with_context(active_test=False).search([('rounding', '<', new_rounding)])
```

Reply:
I would not put a warning in this case. It's confusing if the UoM is archived, he will probably not use it. Would rather check it at write if you are afraid of such a use case.